BookCatalog/Catalog.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Collections;
+
+namespace BookCatalog
+{
+    public class Catalog : IEnumerable<Book>
+    {
+        private Dictionary<string, Book> _books;
+
+        public Catalog()
+        {
+            _books = [];
+        }
+
+        public Book this[string isbn]
+        {
+            get
+            {
+                var bookHelper = new BookHelper();
+
+                if (!bookHelper.IsIsbnInCorrectFormat(isbn))
+                {
+                    throw new ArgumentException("Invalid ISBN format.");
+                }
+
+                return _books[bookHelper.UnifyIsbn(isbn)];
+            }
+        }
+
+        public void Add(Book book)
+        {
+            var bookHelper = new BookHelper();
+
+            if (!_books.ContainsKey(bookHelper.UnifyIsbn(book.Isbn)))
+            {
+                _books.Add(book.Isbn, book);
+            }
+        }
```

Comment:
simplify using LINQ and standard enumerator from books collection

Reply:
Fixed